### ISSUE TITLE
How can I train DEXTR Multiple GPU and Big Batch Size

### ISSUE THREAD

**Issue by Humayun-CF** [scaelles/DEXTR-PyTorch#9]
Hi,

How can I trained DEXTR with multiple GPU either with same batch size (5) or bigger batch size (16 or 32)? What are changes I will do in the code to train on Multiple GPU?

best.
-Humayun

**Comment by kmaninis** [comment]
Hi, you may wanna take a look at [DataParallel](https://pytorch.org/tutorials/beginner/former_torchies/parallelism_tutorial.html#dataparallel). Inside the code, you just need to wrap this class around the model, as in the example. Beware that batchnorm statistics (in case you train the batchnorm layers) are computed per GPU.

**Comment by Humayun-CF** [comment]
Thanks.
One more question, I have 4 GPUs machine and I am training this method using 1 GPU by explicitly mentioning in arguments as gpu_id = 0. 
`device = torch.device("cuda:"+str(gpu_id) if torch.cuda.is_available() else "cpu")
`
When I check with Nvidia-smi, only one process is running on that GPU and no process running on rest of GPU, but memory from all GPU are allocated and there is no free memory in GPUs. 
When I tried to run another training or evaluation program on different GPU but it gave error message there is no cuda memory available. How can I debug it and this code only allocate memory on one GPU instead of all GPUs.

**Comment by scaelles** [comment]
Hello,
When you specify one GPU in the device variable it shouldn't take memory on the others (in theory). In practice, if that's the case, the trick that you can use is always setting gpu_id=0 and then when calling the script use the CUDA_VISIBLE_DEVICES flag so `CUDA_VISIBLE_DEVICES=1 python train.py` for running in GPU 1 for example.
Let us know if that helps!

**Comment by Humayun-CF** [comment]
Thanks It work now.

**Comment by Humayun-CF** [comment]
Thanks. It is working now. 
Best,
